packages/gestalt/src/Link.js

```diff
@@ -144,6 +147,12 @@ const LinkWithForwardRef: AbstractComponent<Props, HTMLAnchorElement> = forwardR
 
   const { isFocusVisible } = useFocusVisible();
 
+  let underlineStyle = inline ? 'always' : 'hover';
+
+  if (underline && underline !== 'auto') {
```

Comment:
You can remove the `underline !== 'auto'` check if you remove the default value in the destructuring above (it's not necessary)

Reply:
@dangerismycat I wanted to add the underline="auto" so it's reflected in the docs, as the natural state of  the 
 component is to manage internally  `underline` (which is basically the `auto`)

Reply:
You can also accomplish that with a comment alongside the prop definition. I think we do this in Box and Flex where the defaults aren't defined in JS